front/pages/api/v1/w/[wId]/apps/[aId]/runs/index.ts

```diff
@@ -264,92 +186,194 @@ async function handler(
         });
       }
 
-      await Run.create({
-        dustRunId: runRes.value.run.run_id,
-        appId: app.id,
-        runType: "deploy",
-        workspaceId: keyRes.value.workspaceId,
-      });
-
-      let run: RunType = runRes.value.run;
-      run.specification_hash = run.app_hash;
-      delete run.app_hash;
-
-      // If `blocking` is set, poll for run completion.
-      if (req.body.blocking) {
-        const runId = run.run_id;
-        try {
-          await poll({
-            fn: async () => {
-              const run = await coreAPI.getRunStatus({
-                projectId: app.dustAPIProjectId,
-                runId,
+      if (req.body.stream) {
```

Comment:
I would maybe start by decideing exactly in which mode we are based on stream/blocking

type RunFlavor = "blocking" | "streaming" | "non-blocking"

That way they are are disjunct and before/after the streaming loop you can switch with an assertNever ?